### DISCUSSION TITLE
IOS roaming from Wifi to LTE "mobike" drop after 8 min

### DISCUSSION THREAD

**Discussion by abeditani** [strongswan/strongswan#521] [Q&A]
**System (please complete the following information):**
 - OS: Ubuntu 20.04
 - Kernel version (if applicable): 5.11.0-25-generic
 - strongSwan version(s): 5.9.1
 - Tested/confirmed with the latest version: [yes/no] no

**Describe the bug**
when switching from wifi to lte the vpn is dropped after 8 mins, the vpn doesnt drop if u stay on wifi and doesnt drop if you stay on lte, only when u switch "mobike"

**To Reproduce**

on Iphone IOS 14.7
1. connect to wifi
2. connect to vpn
3. turn on Cellular data
4. turn off wifi
5. look at the timer when it reaches 8:00 mins the vpn is dropped

**Expected behavior**
vpn is dropped

**Logs/Backtraces**
If applicable, add logs or backtraces to help explain your problem.
[proxyndp.updown.txt](https://github.com/strongswan/strongswan/files/6877226/proxyndp.updown.txt)
[ipsec.conf.txt](https://github.com/strongswan/strongswan/files/6877227/ipsec.conf.txt)

13[NET] sending packet: from Strongswan Public IP[4500] to Iphone Public IP [1210] (80 bytes)
13[ENC] generating INFORMATIONAL response 7 [ ]
13[IKE] IKE_SA deleted
13[IKE] IKE_SA deleted
13[IKE] deleting IKE_SA ikev2-vpn[5] between Strongswan Public IP[PUBLIC DNS NAME]...Iphone Public IP[LOCAL ID]
13[IKE] deleting IKE_SA ikev2-vpn[5] between Strongswan Public IP[PUBLIC DNS NAME]...Iphone Public IP[LOCAL ID]
13[IKE] received DELETE for IKE_SA ikev2-vpn[5]
13[ENC] parsed INFORMATIONAL request 7 [ D ]
13[NET] received packet: from Iphone Public IP [1210] to Strongswan Public IP[4500] (80 bytes)

**Additional context**

**Comment by tobiasbrunner** [comment]
Please post more of the log. But according to what you posted, it might be the client who just deletes the IKE_SA for some reason.

**Reply by abeditani** [reply]
i only have debug enabled at level 1, this is everything related to this client, what level do u want the debug on so i can repost
note: it is known that IOS does a rekey at 8 mins, http://www.openradar.appspot.com/29821241, if that helps and i dont have PFS enabled.
thank you,

**Reply by tobiasbrunner** [reply]
See #196 for recommended log settings.

**Reply by abeditani** [reply]
attached is the log file,

username is abeditani
wifi public ip: 46.152.194.126 
lte public ip: 93.168.102.210

connected to wifi after 5 seconds moved to lte, after 8:00 mins exact drop

[charon_debug1.log](https://github.com/strongswan/strongswan/files/6877691/charon_debug1.log)

**Reply by tobiasbrunner** [reply]
As suspected, the client just deletes the IKE_SA for some unknown reason (there is no attempt to rekey the SA):

    Mon, 2021-07-26, 14:12:43 11[IKE] <ikev2-vpn|5> nothing to initiate
    Mon, 2021-07-26, 14:12:56 14[NET] <ikev2-vpn|5> received packet: from 93.168.102.210[2556] to 157.90.158.35[4500] (80 bytes)
    Mon, 2021-07-26, 14:12:56 14[ENC] <ikev2-vpn|5> parsed INFORMATIONAL request 7 [ D ]
    Mon, 2021-07-26, 14:12:56 14[IKE] <ikev2-vpn|5> received DELETE for IKE_SA ikev2-vpn[5]

You could try getting the client logs (I think that's possible via Xcode), or contact Apple about it. Other than that I don't know if there is anything we can do.

**Reply by abeditani** [reply]
i removed all firewall rules, i tired every possible scenario with no luck

to replicate the problem
- connect to wifi
- connect to vpn
- turn off wifi
- run ipsec stroke rekey XXXX
the connection is dropped with the following error,

rekey is disabled on the server side, but the iphone on min 8 is doing a rekey, the following error is the same every time the vpn is dropped 

please help

SS logs

-------------------------------------------------------------------------------------------------------------
Sat, 2021-07-31, 04:38:14 08[CFG] received stroke: rekey 'ikev2[3]'
Sat, 2021-07-31, 04:38:14 08[IKE] <ikev2|3> queueing IKE_REKEY task
Sat, 2021-07-31, 04:38:14 08[IKE] <ikev2|3> activating new tasks
Sat, 2021-07-31, 04:38:14 08[IKE] <ikev2|3>   activating IKE_REKEY task
Sat, 2021-07-31, 04:38:14 08[IKE] <ikev2|3> IKE_SA ikev2[3] state change: ESTABLISHED => REKEYING
Sat, 2021-07-31, 04:38:14 08[IKE] <ikev2|3> initiating IKE_SA ikev2[4] to 93.168.80.17
Sat, 2021-07-31, 04:38:14 08[IKE] <ikev2|3> IKE_SA ikev2[4] state change: CREATED => CONNECTING
Sat, 2021-07-31, 04:38:14 08[LIB] <ikev2|3> size of DH secret exponent: 2047 bits
Sat, 2021-07-31, 04:38:14 08[CFG] <ikev2|3> configured proposals: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Sat, 2021-07-31, 04:38:14 08[ENC] <ikev2|3> generating CREATE_CHILD_SA request 0 [ SA No KE ]
Sat, 2021-07-31, 04:38:14 08[NET] <ikev2|3> sending packet: from 157.90.158.35[4500] to 93.168.80.17[1344] (432 bytes)
Sat, 2021-07-31, 04:38:15 10[NET] <ikev2|3> received packet: from 93.168.80.17[1344] to 157.90.158.35[4500] (80 bytes)
Sat, 2021-07-31, 04:38:15 10[ENC] <ikev2|3> parsed INFORMATIONAL request 7 [ D ]
Sat, 2021-07-31, 04:38:15 10[IKE] <ikev2|3> received DELETE for IKE_SA ikev2[3]
Sat, 2021-07-31, 04:38:15 10[IKE] <ikev2|3> deleting IKE_SA ikev2[3] between 157.90.158.35[itani.duckdns.org]...93.168.80.17[abeditani]
Sat, 2021-07-31, 04:38:15 10[IKE] <ikev2|3> IKE_SA ikev2[3] state change: REKEYING => DELETING
Sat, 2021-07-31, 04:38:15 10[IKE] <ikev2|3> IKE_SA deleted
Sat, 2021-07-31, 04:38:15 10[IKE] <ikev2|3> detected IKE_REKEY collision with IKE_DELETE
Sat, 2021-07-31, 04:38:15 10[ENC] <ikev2|3> generating INFORMATIONAL response 7 [ ]
Sat, 2021-07-31, 04:38:15 10[NET] <ikev2|3> sending packet: from 157.90.158.35[4500] to 93.168.80.17[1344] (80 bytes)
Sat, 2021-07-31, 04:38:15 10[IKE] <ikev2|4> IKE_SA ikev2[4] state change: CONNECTING => DESTROYING

-------------------------------------------------------------------------------------------------------------

![issue](https://user-images.githubusercontent.com/2196005/127722520-72f4c258-2f27-4632-a315-cebce5700e84.PNG)

**Reply by tobiasbrunner** [reply]
Please contact Apple, there is nothing we can do if their client just deletes the IKE_SA.
